File: app/src/main/java/vn/loitp/app/activity/customviews/menu/residemenu/ResideMenuActivity.java
package vn.loitp.app.activity.customviews.menu.residemenu;

import android.graphics.Color;
import android.os.Bundle;
import android.view.MotionEvent;
import android.view.View;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

import com.annotation.IsFullScreen;
import com.annotation.LayoutId;
import com.annotation.LogTag;
import com.core.base.BaseFontActivity;
import com.core.utilities.LAppResource;
import com.views.menu.residemenu.ResideMenu;
import com.views.menu.residemenu.ResideMenuItem;

import vn.loitp.app.R;

@LayoutId(R.layout.reside_menu)
@LogTag("ResideMenuActivity")
@IsFullScreen(false)
public class ResideMenuActivity extends BaseFontActivity implements View.OnClickListener {
    private ResideMenu resideMenu;
    private ResideMenuItem itemHome;
    private ResideMenuItem itemProfile;
    private ResideMenuItem itemCalendar;
    private ResideMenuItem itemSettings;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setUpMenu();
        if (savedInstanceState == null)
            changeFragment(new HomeFragment());
    }

    private void setupUIResideMenuItem(final ResideMenuItem resideMenuItem) {
        resideMenuItem.setTextColor(Color.BLACK);
        resideMenuItem.setTextShadow(Color.WHITE);
        resideMenuItem.setTextSize(getResources().getDimension(R.dimen.txt_medium));
        resideMenuItem.setIvIconSizeDp(20);
    }

    private void setUpMenu() {
        // attach to current activity;
        resideMenu = new ResideMenu(this);
        resideMenu.getRealtimeBlurView().setBlurRadius(50);
        resideMenu.getRealtimeBlurView().setOverlayColor(LAppResource.INSTANCE.getColor(R.color.black65));
        resideMenu.setUse3D(true);
        resideMenu.setBackground(R.drawable.iv);
        resideMenu.attachToActivity(this);
        resideMenu.setMenuListener(menuListener);
        //valid scale factor is between 0.0f and 1.0f. leftmenu'width is 150dip. 
        resideMenu.setScaleValue(0.6f);

        // create menu items;
        itemHome = new ResideMenuItem(this, R.mipmap.ic_launcher, "Home");
        itemProfile = new ResideMenuItem(this, R.mipmap.ic_launcher, "Profile");
        itemCalendar = new ResideMenuItem(this, R.mipmap.ic_launcher, "Calendar");
        itemSettings = new ResideMenuItem(this, R.mipmap.ic_launcher, "Settings");

        setupUIResideMenuItem(itemHome);
        setupUIResideMenuItem(itemProfile);
        setupUIResideMenuItem(itemCalendar);
        setupUIResideMenuItem(itemSettings);

        itemHome.setOnClickListener(this);
        itemProfile.setOnClickListener(this);
        itemCalendar.setOnClickListener(this);
        itemSettings.setOnClickListener(this);

        resideMenu.addMenuItem(itemHome, ResideMenu.DIRECTION_LEFT);
        resideMenu.addMenuItem(itemProfile, ResideMenu.DIRECTION_LEFT);
        resideMenu.addMenuItem(itemCalendar, ResideMenu.DIRECTION_RIGHT);
        resideMenu.addMenuItem(itemSettings, ResideMenu.DIRECTION_RIGHT);

        // You can disable a direction by setting ->
        // resideMenu.setSwipeDirectionDisable(ResideMenu.DIRECTION_RIGHT);

        findViewById(R.id.title_bar_left_menu).setOnClickListener(view -> resideMenu.openMenu(ResideMenu.DIRECTION_LEFT));
        findViewById(R.id.title_bar_right_menu).setOnClickListener(view -> resideMenu.openMenu(ResideMenu.DIRECTION_RIGHT));
    }

    @Override
    public boolean dispatchTouchEvent(MotionEvent ev) {
        return resideMenu.dispatchTouchEvent(ev);
    }

    @Override
    public void onClick(View view) {

        if (view == itemHome) {
            changeFragment(new HomeFragment());
        } else if (view == itemProfile) {
            changeFragment(new ProfileFragment());
        } else if (view == itemCalendar) {
            changeFragment(new CalendarFragment());
        } else if (view == itemSettings) {
            changeFragment(new SettingsFragment());
        }

        resideMenu.closeMenu();
    }

    private ResideMenu.OnMenuListener menuListener = new ResideMenu.OnMenuListener() {
        @Override
        public void openMenu() {
            showShort("Menu is opened!", true);
        }

        @Override
        public void closeMenu() {
            showShort("Menu is closed!", true);
        }
    };

    private void changeFragment(final Fragment targetFragment) {
        resideMenu.clearIgnoredViewList();
        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.main_fragment, targetFragment, "fragment")
                .setTransitionStyle(FragmentTransaction.TRANSIT_FRAGMENT_FADE)
                .commit();
    }

    // What good method is to access resideMenu？
    public ResideMenu getResideMenu() {
        return resideMenu;
    }
}

File: app/src/main/java/vn/loitp/app/a/customviews/menu/resideMenu/ResideMenuActivity.kt
package vn.loitp.app.a.customviews.menu.resideMenu

import android.graphics.Color
import android.os.Bundle
import android.view.MotionEvent
import android.view.View
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentTransaction
import com.loitp.annotation.IsFullScreen
import com.loitp.annotation.LogTag
import com.loitp.core.base.BaseFontActivity
import com.loitp.core.utilities.LAppResource
import com.loitp.views.menu.resideMenu.ResideMenu
import com.loitp.views.menu.resideMenu.ResideMenuItem
import kotlinx.android.synthetic.main.activity_reside_menu.*
import vn.loitp.R

@LogTag("ResideMenuActivity")
@IsFullScreen(false)
class ResideMenuActivity : BaseFontActivity(), View.OnClickListener {

    var resideMenu: ResideMenu? = null
    private var itemHome: ResideMenuItem? = null
    private var itemProfile: ResideMenuItem? = null
    private var itemCalendar: ResideMenuItem? = null
    private var itemSettings: ResideMenuItem? = null

    override fun setLayoutResourceId(): Int {
        return R.layout.activity_reside_menu
    }

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setUpMenu()
        if (savedInstanceState == null) {
            changeFragment(HomeFragment())
        }
    }

    private fun setupUIResideMenuItem(resideMenuItem: ResideMenuItem) {
        resideMenuItem.setTextColor(Color.BLACK)
        resideMenuItem.setTextShadow(Color.WHITE)
        resideMenuItem.setTextSize(resources.getDimension(R.dimen.txt_medium))
        resideMenuItem.setIvIconSizeDp(20)
    }

    private fun setUpMenu() {
        resideMenu = ResideMenu(this)
        resideMenu?.apply {
            this.realtimeBlurView.setBlurRadius(50f)
            this.realtimeBlurView.setOverlayColor(LAppResource.getColor(R.color.black65))
            this.setUse3D(true)
            this.setBackground(R.drawable.iv)
            this.attachToActivity(this@ResideMenuActivity)
            this.menuListener = menuListener
            // valid scale factor is between 0.0f and 1.0f. leftmenu'width is 150dip.
            this.setScaleValue(0.6f)
        }

        // create menu items;
        itemHome = ResideMenuItem(this, R.drawable.ic_launcher_loitp, "Home")
        itemProfile = ResideMenuItem(this, R.drawable.ic_launcher_loitp, "Profile")
        itemCalendar = ResideMenuItem(this, R.drawable.ic_launcher_loitp, "Calendar")
        itemSettings = ResideMenuItem(this, R.drawable.ic_launcher_loitp, "Settings")

        itemHome?.let {
            setupUIResideMenuItem(it)
        }
        itemProfile?.let {
            setupUIResideMenuItem(it)
        }
        itemCalendar?.let {
            setupUIResideMenuItem(it)
        }
        itemSettings?.let {
            setupUIResideMenuItem(it)
        }

        itemHome?.setOnClickListener(this)
        itemProfile?.setOnClickListener(this)
        itemCalendar?.setOnClickListener(this)
        itemSettings?.setOnClickListener(this)

        resideMenu?.apply {
            this.addMenuItem(itemHome, ResideMenu.DIRECTION_LEFT)
            this.addMenuItem(itemProfile, ResideMenu.DIRECTION_LEFT)
            this.addMenuItem(itemCalendar, ResideMenu.DIRECTION_RIGHT)
            this.addMenuItem(itemSettings, ResideMenu.DIRECTION_RIGHT)
        }

        // You can disable a direction by setting ->
        // resideMenu.setSwipeDirectionDisable(ResideMenu.DIRECTION_RIGHT);
        titleBarLeftMenu.setOnClickListener {
            resideMenu?.openMenu(ResideMenu.DIRECTION_LEFT)
        }
        titleBarRightMenu.setOnClickListener {
            resideMenu?.openMenu(ResideMenu.DIRECTION_RIGHT)
        }
    }

    override fun dispatchTouchEvent(ev: MotionEvent): Boolean {
        return resideMenu?.dispatchTouchEvent(ev) ?: false
    }

    override fun onClick(view: View) {
        when {
            view === itemHome -> {
                changeFragment(HomeFragment())
            }
            view === itemProfile -> {
                changeFragment(ProfileFragment())
            }
            view === itemCalendar -> {
                changeFragment(CalendarFragment())
            }
            view === itemSettings -> {
                changeFragment(SettingsFragment())
            }
        }
        resideMenu?.closeMenu()
    }

//    private val menuListener: OnMenuListener = object : OnMenuListener {
//        override fun openMenu() {
//            showShortInformation("Menu is opened!", true)
//        }
//
//        override fun closeMenu() {
//            showShortInformation("Menu is closed!", true)
//        }
//    }

    private fun changeFragment(targetFragment: Fragment) {
        resideMenu?.clearIgnoredViewList()
        supportFragmentManager
            .beginTransaction()
            .replace(R.id.mainFragment, targetFragment, "fragment")
            .setTransitionStyle(FragmentTransaction.TRANSIT_FRAGMENT_FADE)
            .commit()
    }
}
